python3/amplify/src/lambda_function.py

```diff
@@ -0,0 +1,128 @@
+
+        result = amplify.generate_access_logs(
+            startTime=access_logs_starttime,
+            endTime=access_logs_endtime,
+            domainName=os.getenv('AMPLIFY_DOMAIN'),
+            appId=os.getenv('AMPLIFY_APP_ID')
+        )
+        return result["logUrl"]
+    except Exception as e:
+        logger.warning(
+            f'Error occurred while trying get access url please check your credentials: {e}.')
+
+
+def covert_csv_to_array_of_logs(log_url):
+    json_array = []
+
+    try:
+        response = urllib.request.urlopen(log_url)
+        lines = [l.decode('utf-8') for l in response.readlines()]
+        reader = csv.DictReader(lines, FIELDNAMES)
+
+        for row in reader:
+            json_array.append(json.dumps(row))
+
+        if len(json_array) > 0:
+            json_array.pop(0)
+    except Exception as e:
+        logger.warning(
+            f'Error occurred while trying convert csv to array of logs: {e}.')
+
+    return json_array
+
+
+def _extract_aws_amplify_logs_data(event):
+    # type: (dict) -> dict
+    event_time = datetime.datetime.strptime(
+        event["time"], "%Y-%m-%dT%H:%M:%SZ")
+    try:
+        access_logs_endtime = event_time
+
+        access_logs_starttime = access_logs_endtime - \
+            datetime.timedelta(minutes=5)
+
+        log_url = get_amplify_access_log_link(
+            access_logs_starttime, access_logs_endtime)
+
+        logs = covert_csv_to_array_of_logs(log_url)
+
+        return logs
+
+    except ValueError as e:
+        logger.error("Got exception while loading json, message: {}".format(e))
+        raise ValueError("Exception: json loads")
+
+
+def _add_timestamp(log):
+    # type: (dict) -> dict
+    if '@timestamp' not in log:
+        timestamp_val = log['date'] + 'T' + log['time'] + 'Z'
+        log['@timestamp'] = timestamp_val
+        del log['date']
+        del log['time']
+    return log
+
+
+def _get_additional_logs_data(log, context):
+    # type: (dict, 'LambdaContext') -> dict
+    try:
+        log['function_version'] = context.function_version
+        log['invoked_function_arn'] = context.invoked_function_arn
+    except KeyError:
+        logger.info(
+            'Failed to find context value. Continue without adding it to the log')
+
+    try:
+        log['type'] = os.environ['TYPE']
+    except KeyError:
+        logger.info("Using default TYPE 'logzio_amplify_access_lambda'.")
+        log['type'] = 'logzio_amplify_access_lambda'
+    return log
+
+
+def lambda_handler(event, context):
+    # type (dict, 'LambdaContext') -> None
+
+    aws_logs_data = _extract_aws_amplify_logs_data(event)
+    shipper = LogzioShipper()
+
+    logger.info("About to send {} logs".format(
+        len(aws_logs_data)))
+    for log in aws_logs_data:
+        json_log = json.loads(log)
+
+        if not isinstance(json_log, dict):
+            raise TypeError(
+                "Expected log to be a dict but found another type")
```

Comment:
Not sure about the `if` condition.
`json.loads` will return a dict if the input is valid, and if not will raise an error, so I think this condition is irrelevant.
Also, this way we'll lose the "bad" log, and the logs that are after it, because the error will terminate the lambda function.
I think that a better approach will be to:
1. Remove the current if statement.
2. Wrap the `json.loads` line with a try-except. Then, on the except part - log in the function's logger the error and which log caused the error, and then continue.
That way the function will continue to run for the remaining logs, and at least we'll have the corrupted log documented.